src/async/MuxedInterruptDistributor.cc

```diff
@@ -35,14 +35,19 @@ namespace ChimeraTK::async {
     for(auto const& metaDataEntry : backend.getMetadataCatalogue()) {
       auto const& key = metaDataEntry.first;
       if(key[0] == '!') {
-        auto jkey = nlohmann::json::parse(std::string({++key.begin(), key.end()})); // key without the !
-        auto interruptId = jkey.get<std::vector<size_t>>();
-
-        if(interruptId == subdomainID) {
-          auto jdescriptor = nlohmann::json::parse(metaDataEntry.second);
-          auto controllerType = jdescriptor.begin().key();
-          auto controllerDescription = jdescriptor.front().dump();
-          return {controllerType, controllerDescription};
+        try {
+          auto jkey = nlohmann::json::parse(std::string({++key.begin(), key.end()})); // key without the !
```

Comment:
maybe add a "leading" before ! in the comment

Reply:
I like that. I put it in.